src/property.ts

```diff
@@ -115,9 +128,9 @@ class Property {
                     this.cast = util.castNullableLink;
                 }
             }
-        }
-        if (this.choices && this.cast) {
-            this.choices = this.choices.map(choice => this.cast(choice));
+        } else if (this.choices) {
+            const castFunc: (arg: unknown) => any = this.cast;
```

Comment:
yup, I couldn't get the types to work otherwise

Reply:
if `this.cast` can't be undefined by this point, then type for cast may need to be changed to reflect that. if it can, then I would just change the condition to be `} else if (this.choices && this.cast) {`

if it can be undefined but you know it's not in this particular situation for some reason and don't want a redundant check, then you could do
```ts
this.choices = this.choices.map((choice) => (this.cast as NonNullable<typeof this.cast>)(choice));
```

Reply:
done